pkg/common/common.go

```diff
@@ -299,3 +303,48 @@ func ApplyUnsupportedServerArgsOverride(container *corev1.Container, unsupported
 	}
 	return nil
 }
+
+// UpdateBackupStorageLocation updates the BackupStorageLocation spec and config.
+func UpdateBackupStorageLocation(bsl *velerov1.BackupStorageLocation, bslSpec velerov1.BackupStorageLocationSpec) error {
```

Comment:
I am aware that currently this function does not really return error, however it's a design to ensure if it does, both NABSL controller and bsl functions from the OADP project does not need to update calls to it.

I can imagine some validation could be done here which would return error.

Reply:
ok so expect nolint here in the future from some linter

Reply:
yes (maybe, or we add err validation first) :)